client/src/pages/EditProfilePage/EditProfilePage.tsx

```diff
@@ -14,33 +15,117 @@ const EditProfilePage = () => {
   const userID = useAppSelector((state) => state.user.userData?._id);
   const fileInputRef = useRef<HTMLInputElement>(null);
 
-  const [formData, setFormData] = useState({
+  type FormDataType = {
+    email: string;
+    nickName: string;
+    cohort: string;
+    personalBio: string;
+    currentPositionTitle: string;
+    currentPositionCompany: string;
+    availabilityForNetworking: boolean;
+    linkedInProfile: string;
+    gitHubProfile: string;
+    twitter: string;
+    blog: string;
+    skills: string[];
+    specializations: string[];
+  };
+
+  const [formData, setFormData] = useState<FormDataType>({
     email: '',
+    nickName: '',
+    cohort: '',
     personalBio: '',
+    currentPositionTitle: '',
+    currentPositionCompany: '',
+    availabilityForNetworking: false,
+    linkedInProfile: '',
+    gitHubProfile: '',
+    twitter: '',
+    blog: '',
+    skills: [] as string[],
+    specializations: [] as string[],
   });
 
+  const [skillInput, setSkillInput] = useState('');
+
   const [file, setFile] = useState<File | null>(null);
 
   useEffect(() => {
     if (userID) dispatch(fetchUserProfile(userID as string));
-  }, [dispatch]);
+  }, [dispatch, userID]);
```

Comment:
Wondering if userID in the dependency array makes a difference? I'm thinking there should only ever be a userID in redux state if the user is logged in. If they're not logged in they shouldn't have access to this page / get booted out to the login page. Same for the check on line 55

Reply:
Linting :/

Reply:
💀

Reply:
🤣